internal/standby/standby.go

```diff
@@ -75,13 +76,61 @@ func (s *Service) RunMQTT(ctx context.Context) error {
 			}
 		})
 
+	s.subscribeToTopic(s.cfg.MQTT.ReadCommandTopic,
+		func(client mqtt.Client, msg mqtt.Message) {
+			s.logger.Debug(fmt.Sprintf("Received message: %s from topic: %s", msg.Payload(), msg.Topic()))
+			s.latestCommandReceived = time.Now()
+		})
+
 	return nil
 }
 
 func (s *Service) StopMQTT() {
 	s.MQTTClient.Disconnect(uint(1000))
 }
 
+func (s *Service) RunDetector(ctx context.Context) {
+	checkInterval := time.Duration(s.cfg.Standby.CheckInterval) * time.Second
+	outageDuration := time.Duration(s.cfg.Standby.OutageInterval) * time.Second
+
+	// initialise this so we don't immediately go into failure mode on startup
+	s.latestCommandReceived = time.Now()
```

Comment:
can we insted inject the time? as part of the signature of `RunDetector`? it makes testing things way easier

Reply:
Do you mean inject the current time when calling `RunDetector` to initialise this value? Sure, or I could even initialise it when instantiating the service - it just needs to have some recent value before the first run of `checkForOutage`